src/ape_ethereum/ecosystem.py

```diff
@@ -10,8 +21,92 @@
 }
 
 
+class Transaction(TransactionAPI):
+    def is_valid(self) -> bool:
+        return False
+
+    def as_dict(self) -> dict:
+        data = super().as_dict()
+
+        # Clean up data to what we expect
+        data.pop("chain_id")
+        data.pop("sender")
+        data["to"] = data.pop("receiver")
+        data["gas"] = data.pop("gas_limit")
+        data["gasPrice"] = data.pop("gas_price")
+
+        # NOTE: Don't publish signature
+        data.pop("signature")
+
+        return data
+
+    def encode(self) -> bytes:
+        data = self.as_dict()
+        unsigned_txn = serializable_unsigned_transaction_from_dict(data)
+        return encode_transaction(
+            unsigned_txn,
+            (
+                to_int(self.signature[:1]),
+                to_int(self.signature[1:33]),
+                to_int(self.signature[33:65]),
+            ),
+        )
+
+
+class Receipt(ReceiptAPI):
+    @classmethod
+    def decode(cls, data: dict) -> ReceiptAPI:
+        return cls(  # type: ignore
+            txn_hash=data["hash"],
+            status=TransactionStatusEnum(data["status"]),
+            block_number=data["blockNumber"],
+            gas_used=data["gasUsed"],
+            gas_price=data["gasPrice"],
+            logs=data["logs"],
+            contract_address=data["contractAddress"],
+        )
+
+
 class Ethereum(EcosystemAPI):
-    pass
+    transaction_class = Transaction
+    receipt_class = Receipt
+
+    def encode_deployment(
+        self, deployment_bytecode: bytes, abi: Optional[ABI], *args, **kwargs
+    ) -> Transaction:
+        txn = Transaction(**kwargs)  # type: ignore
+        txn.data = deployment_bytecode
+
+        # Encode args, if there are any
+        if abi and abi.inputs:
+            input_types = [i.canonical_type for i in abi.inputs]
+            txn.data += abi_encode(input_types, *args)
+
+        return txn
+
+    def encode_transaction(self, abi: ABI, *args, **kwargs) -> Transaction:
+        txn = Transaction(**kwargs)  # type: ignore
+
+        # Add method ID
+        txn.data = keccak(to_bytes(text=abi.selector))[:4]
+
+        # Encode args, if there are any
+        if abi.inputs:
+            input_types = [i.canonical_type for i in abi.inputs]
+            txn.data += abi_encode(input_types, *args)
+
+        else:
+            assert len(args) == 0
 
+        return txn
 
-# TODO: Define VM-specific stuff here
+    def decode_event(self, abi: ABI, receipt: "ReceiptAPI") -> "ContractLog":
+        filter_id = keccak(to_bytes(text=abi.selector))
+        raw_event = next(log for log in receipt.logs if log["filter_id"] == filter_id)
+        input_names = [i.name for i in abi.inputs]
+        input_type = ",".join(i.canonical_type for i in abi.inputs)
+        event_data = abi_decode(input_type, raw_event)
+        return ContractLog(  # type: ignore
+            name=abi.name,
+            inputs={k: v for k, v in zip(input_names, event_data)},
```

Comment:
Using `dict(zip(...))` also works and is shorter:

```
>>> x, y = ['a', 'b', 'c'], [1, 2, 3]
>>> {k: v for (k, v) in zip(x, y)}
{'a': 1, 'b': 2, 'c': 3}
>>> dict(zip(x, y))
{'a': 1, 'b': 2, 'c': 3}
```

Reply:
oh yeah duh